samtranslator/open_api/open_api.py

```diff
@@ -461,7 +461,9 @@ def _set_method_authorizer(self, path, method_name, authorizer_name, authorizers
             authorization_scopes = []
 
         for method_definition in self.iter_on_method_definitions_for_path_at_method(path, method_name):  # type: ignore[no-untyped-call]
-            existing_security = method_definition.get("security", [])
+            existing_security = method_definition.get("security")
+            if not existing_security:
+                existing_security = []
```

Comment:
why? Aren't they doing the same thing?

Reply:
see the description above, if `security: null`, it won't return the default value. The default value is for the key doesn't exist

Reply:
maybe add a comment. it's not clear if one doesn't know/remember that `security: null` can be an actual pair

Reply:
And also, should we add a test for that?

Reply:
I changed the implementation and it now raises an exception

Reply:
one test added